portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -44,6 +47,7 @@
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
 
     int requestLimit = Integer.parseInt(request.getParameter("request-limit"));
+    String language = request.getParameter("language");
```

Comment:
The value that comes in here should be considered "user input". It's trivial to send a request for any value, so it's important that the input is validated to ensure it's sane.

Reply:
Would a sensible solution be to check that it is included in the language list mentioned above?

Reply:
Yes. For any "real" code, any user input should be processed by some kind of sanitisation function, which will only give back a cleaned, validated, usable result. (Obligatory XKCD: https://xkcd.com/327/, and the corresponding Explain: https://www.explainxkcd.com/wiki/index.php/Little_Bobby_Tables)